torch/_dynamo/variables/builtin.py

```diff
@@ -1459,10 +1459,21 @@ def call_hasattr(self, tx, obj, attr):
                 return variables.ConstantVariable(hasattr(obj.fn, name))
             return obj.call_hasattr(tx, name)
 
-    def call_map(self, tx, fn, seq):
-        if seq.has_unpack_var_sequence(tx):
-            items = [fn.call_function(tx, [x], {}) for x in seq.unpack_var_sequence(tx)]
-            return variables.TupleVariable(items)
+   def call_map(self, tx, fn, *seqs):
+        if all(seq.has_unpack_var_sequence(tx) for seq in seqs):
+            unpacked = [seq.unpack_var_sequence(tx) for seq in seqs]
+
+            if not all(len(curr) == len(unpacked[0]) for curr in unpacked):
+                raise UserError(
+                    ValueError,
+                    "map() has atleast one argument of len differing from others",
+                )
+            items = []
+            for i in range(len(unpacked[0])):
+                params = []
+                for j in range(len(unpacked)):
+                    params.append(unpacked[j][i])
```

Comment:
```suggestion
                params = list(unpacked[j])
```

Reply:
A couple of doubts here, 
1. What is the value of `j`, if we remove the inner loop?
2. Each list in `unpacked` contains at least one value for each required parameter. For example, if we have two parameters, x and y, `unpacked` includes all the variable trackers for both. unpacked[j] would give all the values for parameter `j`, instead we need to use 'unpacked[j][0]' for x and 'unpacked[j][1]' for y. So, how'd `list(unpacked[j])` work? 

An example, 
```
unpacked[0]= [LazyVariableTracker(), LazyVariableTracker(), LazyVariableTracker()] <- x values
unpacked[1]= [LazyVariableTracker(), LazyVariableTracker(), LazyVariableTracker()] <- y values
We need one of each at a time.
```

Reply:
I don't think this is the right suggestion.

Reply:
Thanks for confirming